tests/phpunit/includes/testcase-multisites.php

```diff
@@ -0,0 +1,312 @@
+<?php
+
+/**
+ * Test case offering a standardized way to test blogs in multisite.
+ */
+abstract class PLL_Multisites_TestCase extends WP_UnitTestCase {
+	/**
+	 * Blog in plain permalinks without Polylang.
+	 *
+	 * @var WP_Site
+	 */
+	protected static $blog_without_pll_plain_links;
+
+	/**
+	 * Blog in pretty permalinks without Polylang.
+	 *
+	 * @var WP_Site
+	 */
+	protected static $blog_without_pll_pretty_links;
+
+	/**
+	 * Blog in plain permalinks with Polylang (i.e. default installation).
+	 * Created languages are English and French.
+	 *
+	 * @var WP_Site
+	 */
+	protected static $blog_with_pll_default_links;
+
+	/**
+	 * Blog in pretty permalinks with Polylang and language as directory.
+	 * Created languages are English and French.
+	 *
+	 * @var WP_Site
+	 */
+	protected static $blog_with_pll_directory; // Main blog.
+
+	/**
+	 * Blog in pretty permalinks with Polylang and language as domains.
+	 * Created languages are English and German.
+	 *
+	 * @var WP_Site
+	 */
+	protected static $blog_with_pll_domains;
+
+	/**
+	 * Pretty permalinks structure.
+	 *
+	 * @var string
+	 */
+	protected $pretty_structure = '/%postname%/';
+
+	/**
+	 * Plain permalinks structure.
+	 *
+	 * @var string
+	 */
+	protected $plain_structure = '';
+
+	/**
+	 * Languages data for their creation keyed by language slug.
+	 *
+	 * @var array
+	 */
+	protected $languages = array(
+		'en' => array(
+			'name'       => 'English',
+			'slug'       => 'en',
+			'locale'     => 'en_US',
+			'rtl'        => 0,
+			'flag'       => 'us',
+			'term_group' => 0,
+		),
+		'fr' => array(
+			'name'       => 'Français',
+			'slug'       => 'fr',
+			'locale'     => 'fr_FR',
+			'rtl'        => 0,
+			'flag'       => 'fr',
+			'term_group' => 1,
+		),
+		'de' => array(
+			'name'       => 'Deutsch',
+			'slug'       => 'de',
+			'locale'     => 'de_DE',
+			'rtl'        => 0,
+			'flag'       => 'de',
+			'term_group' => 2,
+		),
+	);
+
+	public function set_up() {
+		parent::set_up();
+
+		// Create all sites.
+		$factory = $this->factory();
+
+		self::$blog_with_pll_directory = get_site( 1 );
```

Comment:
Why using static properties if they are not used in static methods?
Here 4 sites are created per test (so 24 for all 6 tests). Is it intentional or did  you want to create 4 tests per test class?

Reply:
Static to instance props in [7a59368](https://github.com/polylang/polylang/pull/1352/commits/7a593681cd2fc84904eed42625320f728ce60bcb).
I don't know about the original intent, but the sites are deleted in `tear_down()`.